docs/src/main/paradox/getting-started.md

```diff
@@ -0,0 +1,272 @@
+To initialize the Source Provider we need to set parameters to choose the Akka Persistence plugin (Cassandra) to use as well as the name of the tag used for events we're interested in from the journal.
+
+Scala
+:  @@snip [ShoppingCartApp.scala](/examples/src/test/scala/docs/guide/ShoppingCartApp.scala) { #guideSourceProviderSetup }
+
+## Part 2: Build a Stateful Projection handler
+
+While building a projection there are several non-functional requirements to consider.
+_What technology to project into? What message delivery semantics are acceptable for the system? Is it compatible with the chosen Source Provider to enable exactly-once message delivery? Does runtime state need to be maintained in the projection while it's running?_
+It's up to the user to choose the right answers to these questions, but you must research if the answers to these questions compatible with each other.
+
+For this guide we will create a Projection that represents all checked out shopping carts and all the items that were purchased.
+For the purpose of this guide we will persist our Projection to Cassandra with at-least-once semantics.
+The Projection itself will be represented as a Cassandra table.
+
+To proceed we must add the Cassandra Projection library to our project:
+
+@@dependency [sbt,Maven,Gradle] {
+  group=com.lightbend.akka
+  artifact=akka-projection-cassandra_$scala.binary.version$
+  version=$project.version$
+}
+
+Add the following imports to your project:
+
+Scala
+:  @@snip [ShoppingCartApp.scala](/examples/src/test/scala/docs/guide/ShoppingCartApp.scala) { #guideProjectionImports }
+
+It's the user's responsibility to implement the means to project into the target system, the Projection itself will only manage the persistence of the offset (though it is possible to enlist your projection into transactions when using projection implementations that support exactly-once like the @ref:[JDBC](jdbc.md)).
+This guide encapsulates its data access layer in a Repository called the `DailyCheckoutProjectionRepository`.
+The repository will manage two Cassandra tables:
+
+1. The `daily_checkouts` table will contain the actual Projection data of checked out shopping carts.
+Each row in the table will include a date, shopping cart id, item id, and item quantity.
+End users will be able to query the table by a date, cart id, or item id.
+2. The `cart_state` table is an intermediate table that represents the current state of all active shopping carts.
+This is required so that we can restore our state after the projection has been shutdown (i.e. a previously planned shutdown, a failure, or a rebalance).
+This table can also be considered a Projection, even though we are only using it to represent intermediate state in this guide.
+
+Scala
+:  @@snip [ShoppingCartApp.scala](/examples/src/test/scala/docs/guide/ShoppingCartApp.scala) { #guideProjectionRepo }
```

Comment:
It'd be great to point out that the `addCheckout` and `updateCartState` methods in the repository (and how they are used in the handler) are idempotent. Also, indicate why idempotence is important considering this example is using `at-least-once` semantics.

Reply:
#351